### ISSUE TITLE
Spotify Auth Failed

### ISSUE THREAD

**Issue by EsharkyTheGreat** [Rigellute/spotify-tui#1049]
After entering the client id and client secret, the authentication url is opened in my browser and the redirected to `localhost:8888/callback?code=something_here`, however there it is unable to establish a connection and spt prints out `Spotify Auth Failed`

**Comment by pehlicd** [comment]
Have you done this step **4** in `Connecting to Spotify’s API` [doc](https://github.com/Rigellute/spotify-tui#connecting-to-spotifys-api)? You must enter the url in your spotify app dashboard for once.

**Comment by EsharkyTheGreat** [comment]
Yes, I've added the url and as I said I get redirected to localhost after authentication, I even get the code parameter in the url I'm getting redirected to, but I get spotfiy auth failed after that

**Comment by pehlicd** [comment]
Could you go to [Spotify Dashboard](https://developer.spotify.com/dashboard/applications) and go to the app that you created earlier after that click to the **Edit Settings** button then check the **Redirect URIs** section? You should see `http://localhost:8888/callback` url if you don't see, add it and run `spt`.

**Comment by EsharkyTheGreat** [comment]
Yup, as I said already the url is present in the Refirect URIs section, and I also get redirected to localhost:8888/callback?code=<something> just for a second then the spt binary throws auth failed.

(edit) From the code I see that the err message `spotify auth failed` is thrown when the `code` value recieved by the local webserver is used to get a accesstoken but it fails

**Comment by pehlicd** [comment]
I guess it should only be `http://localhost:8888/callback` that url not the `localhost:8888/callback?code=`. You can also check the `${HOME}/.config/spotify-tui/client.yml` file just to be sure that you have provided the true credentials.
